lib/tasks/taxonomy/find_nested_tags.rake

```diff
@@ -0,0 +1,55 @@
+require 'gds_api/rummager'
+require 'gds_api/content_store'
+require 'set'
+
+namespace :taxonomy do
+  desc "Find any content tagged at one taxon level, then tagged again at one of those taxon's ancestors"
+  task :find_nested_tags, [:root_taxon_base_path] => [:environment] do |_, args|
+    root_taxon_base_path = args[:root_taxon_base_path]
+
+    content_store = GdsApi::ContentStore.new(Plek.find('content-store'))
+
+    root_taxon = content_store.content_item(root_taxon_base_path).to_h
+
+    errors = find_content_tagged_at_multiple_levels(root_taxon)
+
+    puts "\n\n"
+    errors.each { |error| puts error }
+  end
+
+  def find_content_tagged_at_multiple_levels(taxon, content_already_tagged = {})
+    taxon_base_path = taxon['base_path']
+
+    taxon_content = GdsApi::Rummager.new(Plek.find('rummager')).search(
+      filter_taxons: [taxon['content_id']],
+      fields: ['link'],
+      count: 1000,
+    )['results']
+
+    errors = []
+
+    taxon_content.each do |content|
+      content_base_path = content['link']
+
+      if content_already_tagged.key?(content_base_path)
+        errors << "#{content_base_path} tagged at #{taxon_base_path}, but also tagged at ancestors: #{content_already_tagged[content_base_path].to_a}"
+      else
+        content_already_tagged[content_base_path] = Set.new
```

Comment:
You can avoid doing this by initializing the hash using `Hash.new { |h, k| h[k] = Set.new }` rather than `{}`.